src/etc/inc/interfaces.inc

```diff
@@ -6000,12 +6005,13 @@ function get_interface_vendor_mac($interface) {
 	global $g;
 
 	$mac = "";
-	$mac_file = "{$g['cf_conf_path']}/vendor_mac_{$interface}";
+	$mac_file = "{$g['vardb_path']}/vendor_mac";
 	if (file_exists($mac_file)) {
-		$mac = file_get_contents($mac_file);
-		if (!is_macaddr($mac)) {
-			$mac = "";
-		}
+		$vendor_mac_contents = file_get_contents($mac_file);
+		$mac_pattern = '(?:[[:xdigit:]]{2}:){5}[[:xdigit:]]{2}';
+		$pattern = '/^'.$interface.': ('.$mac_pattern.')$/m';
```

Comment:
Been looking at this.  But don't think it is just a simple single operation.  So far what I've come up with is this.  Maybe you have better way?
```
$vendor_mac_lines = explode("\n",$vendor_mac_contents);

foreach ($vendor_mac_lines as $line) {
	$line = explode("=",$line);
	$iface = $line[0];
	$mac = $line[1];
	$vendor_mac[$iface] = $mac;
}

$mac = (is_macaddr($vendor_mac[$interface])) ? $vendor_mac[$interface] : "";
```